Entity/Components/HealthComponent.gd

```diff
@@ -17,16 +17,16 @@ func _ready() -> void:
 # If caster is null, we assume that the damage came from an unknown source,
 # so status won't calculate.
 # Use a negative damage value if you want to heal.
-func deal_damage(damage_data: DealDamageData) -> void:
-	var damage: float = damage_data.damage
+func deal_damage(incoming_damage, by_caster) -> void:
+	var damage: float = incoming_damage
```

Comment:
i don't think you need to create a new variable for that

Reply:
True.